src/stores/SessionStore.js

```diff
@@ -0,0 +1,61 @@
+import dis from '../dispatcher';
+import EventEmitter from 'events';
+
+/**
+ * A class for storing application state to do with the session. This is a simple flux
+ * store that listens for actions and updates its state accordingly, informing any
+ * listeners (views) of state changes via the 'update' event.
+ */
+class SessionStore extends EventEmitter {
+    constructor() {
+        super();
+
+        // Initialise state
+        this._state = {
+            cachedPassword: localStorage.getItem('mx_pass'),
+        };
+
+        dis.register(this._onAction.bind(this));
+    }
+
+    _update() {
+        // Persist state to localStorage
+        if (this._state.cachedPassword) {
+            localStorage.setItem('mx_pass', this._state.cachedPassword);
```

Comment:
Yep, in practice `localStorage` is used throughout the app directly and this doesn't gel with a React-based thing. In an ideal world any changes to a `localStorage` thing should first go through a flux store so that the changes can be propagated nicely and the React component question should have no idea that `localStorage` is being used. `Lifecycle` could be `LifecycleStore` that stores and persists lifecycle-related state.

TBF the change to use `SessionStore` is a massive improvement on accessing `mx_pass` directly from `MatrixChat` etc.